internal/retry/policyprovider.go

```diff
@@ -26,44 +26,53 @@ import (
 	"go.uber.org/yarpc/api/transport"
 )
 
+// PolicyProvider returns a retry policy to use for the given context and
+// request.  Nil responses will be interpreted as "no retries".
+type PolicyProvider interface {
+	Policy(context.Context, *transport.Request) *Policy
+}
+
 type serviceProcedure struct {
 	Service   string
 	Procedure string
 }
 
-// procedurePolicyProvider is a new PolicyProvider that
-// has the ability to convert a context and transport request to
-// determine which retry policy to use.
-// The PolicyProvider has the ability to register policies based
-// on service and procedure attributes.  It also has the ability
-// to specify the default retry policy.
-type procedurePolicyProvider struct {
+// ProcedurePolicyProvider is a PolicyProvider that has the ability to resolve
+// retry policies based on the Service and Procedure of each retry request. It
+// also supports returning a default policy if no Service or Procedure matched.
+type ProcedurePolicyProvider struct {
```

Comment:
This is more complicated than just service-procedure mapping. The documentation doesn't state that you're actually supporting multiple levels of precedence here: most specific (service-procedure) to least specific (default). Can you clarify that?